### PR TITLE
Ncnews 2

### PR DESCRIPTION
**Pull request by Joelymodevs**
completed review changes

### CODE REVIEW

File: controllers.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+exports.fetchTopics = (req, res) => {
+    getTopics().then((result) => {
+        res.status(200).send(result)
+    }).catch(err => {
+        res.status(500).send(err)
+    })
+}
+
+exports.fetchEndpoints = (req, res) => {
+    const endpoints = require('./endpoints.json')
```
Comment (pippaamy) [inline controllers.js]: lets pop this require at the top of the page

File: models.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+
+
+exports.getTopics = () => {
+    return db.query('SELECT * FROM topics').then((results) => {
+        return results.rows
+    })
+}
+
+exports.getEndpoints = () => {
```
Comment (pippaamy) [inline models.js:13]: we don't need a model for the endpoints, lets pop this functionality in the controller!

File: controllers.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    res.status(200).send(endpoints).catch(err => {
+        res.status(500).send(err)
```
Comment (pippaamy) [inline controllers.js]: we want to have separate error middleware - not within our controller

File: controllers.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+const { getTopics, getEndpoints} = require('./models')
+
+
+exports.fetchTopics = (req, res) => {
+    getTopics().then((result) => {
+        res.status(200).send(result)
+    }).catch(err => {
+        res.status(500).send(err)
```
Comment (pippaamy) [inline controllers.js]: same here

File: __tests__/endpoints.test.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        });
+      });
+  });
+  test("length expectation", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then((response) => {
+        response.body.forEach((topic) => {
+          expect(Object.keys(topic).length).toBe(2);
```
Comment (pippaamy) [inline __tests__/endpoints.test.js:59]: we want to check the length before the forEach

**Review by pippaamy — COMMENTED** [review]
Good job, when we are resubmitting work we want to be putting the changes on the same pull request! so your changes to topics/task3 should be on the previous pull request as it make it difficult to read otherwise! for now could you make the changes and pop them on this same pull request and pop in another pr on slack :)

File: errorhandles.js [hunk 6]
```diff
@@ -1,8 +1,10 @@
...
-    }
+    console.log(err);
+    res.status(500).send({msg : 'Internal Server Error :('});
+
+}
+exports.handle500 = (err, res , req , next) => {
+    console.log(err);
+    res.status(500).send({msg : 'Internal Server Error :('});
 
 }
```
Comment (robert112233) [inline errorhandles.js:5]: It looks here as if you've got two identical 500 handlers, this shouldn't be necessary

**Review by robert112233 — APPROVED** [review]
Remove the duplicate 500 handler (if I'm seeing that correctly) and merge without PR, well done

**Pull request merged**